OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -291,11 +292,79 @@ public override bool HandleKeyPress(KeyInput e)
 					return ToggleMute();
 				else if (key == Game.Settings.Keys.TogglePlayerStanceColorsKey)
 					return TogglePlayerStanceColors();
+				else if (key == Game.Settings.Keys.NextTrack)
```

Comment:
You should be use the methods exposed on MusicPlaylist, which know how to handle these special cases.  See `MusicPlayerLogic` for an example of how to do this.